worker/search.js

```diff
@@ -60,11 +60,25 @@ async function _indexItem (item, { models }) {
 
   itemcp.wvotes = itemdb.weightedVotes - itemdb.weightedDownVotes
 
+  const bookmarkedBy = await models.bookmark.findMany({
+    where: { itemId: Number(item.id) },
+    select: { userId: true, createdAt: true },
+    orderBy: [
+      {
+        createdAt: 'desc'
+      }
+    ]
+  })
+  itemcp.bookmarkedBy = bookmarkedBy.map(bookmark => bookmark.userId)
+
+  // Use the later of when the item was updated, or when it was last bookmarked, to determine the latest version of the indexed version
+  const latestUpdatedAt = Math.max(new Date(item.updatedAt).getTime(), bookmarkedBy[0] ? new Date(bookmarkedBy[0].createdAt).getTime() : 0)
+
```

Comment:
If an item has `updated_at = 1`, then is bookmarked with a bookmark of `created_at = 2`, `latestUpdatedAt` is set to 2.

When that bookmark is deleted `latestUpdatedAt` is set to 1 because the item has `updated_at = 1` still.

`external_gte` means update the index iff `version` is greater than or equal to the currently stored `version`. Meaning, when the bookmark is deleted, it won't be reindexed because it'll use the lesser `item.updated_at`

--------------

One way to solve this is to remove the trigger on `Bookmark` and simply update `Item.updated_at = now()`, ie "touch" it, when a bookmark is added/removed which will run the `Item` trigger to reindex itself.

There might be another way to do this, but I can't think of one.

Reply:
Good catch! Let me noodle on this

Reply:
We could probably have a special triggered job that is for "removed bookmark on item", which sets the "updated at" on the indexing to be Date.now(). That way we don't touch the item itself

_thinking aloud_

Reply:
That would work. It depends on the server clocks being synchronized but in practice (at least ours for now) that's a fair assumption.

I can't tell if it's more or less hacky than `touch`ing items.

Reply:
> It depends on the server clocks being synchronized but in practice (at least ours for now) that's a fair assumption.

I think this is only a real concern if a bookmark were to be removed immediately before/after an actual edit of the item, right? Otherwise, a little bit of clock skew isn't a big deal since there's usually a large gap in time between item edits (first 10 minutes) and users (un-)bookmarking items forever into the future? Or perhaps I am misunderstanding the point you're making.

> I can't tell if it's more or less hacky than touching items.

Personally I think it's better than `touch`ing the item, because `touch`ing the item suggests the OP of the item updated it, when in reality they did not. So this kinda muddies the history of the item and its audit trail.

At least this way, this is just a false update to facilitate updating search indices 🤷 

I'm working on the changes, I think it's not too bad. But I'll be eager to hear your thoughts once I push the updates :)

Reply:
OK, PR is updated. Let 'er rip!